cpp/include/raft/neighbors/detail/ivf_pq_compute_similarity-inl.cuh

```diff
@@ -448,11 +456,18 @@ __global__ void compute_similarity_kernel(uint32_t n_rows,
       OutT score = kDummy;
       bool valid = i < n_samples;
       if (valid) {
-        score = ivfpq_compute_score<OutT, LutT, vec_t, PqBits>(
-          pq_dim,
-          reinterpret_cast<const vec_t::io_t*>(pq_thread_data),
-          lut_scores,
-          early_stop_limit);
+        // Use the filter.
+        // The purpose of the filter is to decide whether a certain sample acceptable
+        // for a given query.
+        // Technically, a more sophisticated filter may be used that takes
+        // the computed score into account.
+        if (sample_filter(query_ix, label, i)) {
```

Comment:
Are you sure we must have `query_ix` as a parameter of the filter? We either need to remove this parameter or make sure we add the offset into the current batch  https://github.com/rapidsai/raft/blob/da7538c650d9f99495a83d961e845b033b7a20c4/cpp/include/raft/neighbors/detail/ivf_pq_search.cuh#LL833C23-L833C40  (offset_q + offset_b)

Reply:
Definitely, `offset_q + offset_b` needs to be propagated! I'll update the PR.